### PR TITLE
Refactor End-to-End bash testing scripts

### PR DESCRIPTION
**Pull request by electrocucaracha**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind bug

/kind cleanup

> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Most of the testing bash function use similar values, using default arguments can facilitate the readability and reduce errors. This change refactor the testing bash functions and reduces some complexity on them as well as fixes some linting issues.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: e2e/tests/006.sh [hunk 1]
```diff
@@ -23,24 +23,19 @@ export E2EDIR=${E2EDIR:-$HOME/test-infra/e2e}
...
-k8s_wait_exists "$kubeconfig" 600 "default" "packagevariant" "regional-free5gc-amf-regional-free5gc-amf"
-k8s_wait_ready "$kubeconfig" 600 "default" "packagevariant" "regional-free5gc-amf-regional-free5gc-amf"
-k8s_wait_exists "$cluster_kubeconfig" 600 "free5gc-cp" "deployment" "amf-regional"
-k8s_wait_ready_replicas "$cluster_kubeconfig" 600 "free5gc-cp" "deployment" "amf-regional"
+k8s_wait_ready "packagevariant" "regional-free5gc-amf-regional-free5gc-amf"
+k8s_wait_ready_replicas "deployment" "amf-regional" "$cluster_kubeconfig" "free5gc-cp"
 
 # check the SMF
-k8s_wait_ready "$kubeconfig" 600 "default" "packagevariant" "regional-free5gc-smf-regional-free5gc-smf"
-k8s_wait_exists "$kubeconfig" 600 "default" "packagevariant" "regional-free5gc-smf-regional-free5gc-smf"
```
Comment (electrocucaracha) [inline e2e/tests/006.sh:44]: Btw, these instructions should need to be in reversed order.
```suggestion
k8s_wait_exists "$kubeconfig" 600 "default" "packagevariant" "regional-free5gc-smf-regional-free5gc-smf"
k8s_wait_ready "$kubeconfig" 600 "default" "packagevariant" "regional-free5gc-smf-regional-free5gc-smf"
```

**Comment by johnbelamaric** [comment]
Nice clean up! One more thing - since we now put all the clusters in the kubeconfig as contexts, maybe we should change the k8s.sh to use contexts instead of generating them with k8s_get_capi_kubeconfig all the time?

**Comment by electrocucaracha** [comment]
> Nice clean up! One more thing - since we now put all the clusters in the kubeconfig as contexts, maybe we should change the k8s.sh to use contexts instead of generating them with k8s_get_capi_kubeconfig all the time?

That change will affect another functions that are expecting kubeconfig files as arguments, I think that's doable, but it will have bigger impact. The another thing is to include a cache portion to store kubeconfig files.

**Pull request merged**